### PR TITLE
Adding Ambient lighting to transparent surfaces

### PR DESCRIPTION
**Pull request by samcake**
Transparent surfaces weren;t getting the ambient light contribution proeprly so far,
THis pr, simply make the transparent shader use the same code path as opaque for the key lighting.
THis should solve the problem.
Note that infront Overlay3D (transparent or not) are not ambient lighted.

- MOved the AMbient SHperical harmonics in the light direclty

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: a740f4a6de14dc02b0a9c688c6f59eae5f0741e1 available at: https://gist.github.com/504b718a5016eb7d48c6<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7114.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7114.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4c82e0b2b18ccbabb2e842f647d15a6cea7ef481 available at: https://gist.github.com/504b718a5016eb7d48c6<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7114.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7114.dmg

**Comment by OmegaHeron** [comment]
@samcake - Do you have a model reference we might use to test this or a reachable location?  I have a few things with transparency, but, can't really see any difference between production build and PR.  It may be I simply don't know what I'm looking for.

**Comment by samcake** [comment]
RIght, i should have pointed out a model.
You can try this model:
https://hifi-public.s3.amazonaws.com/sam/models/mitsuba/mitsuba_transparent.fbx

and see the difference in the rendering
with this PR, the transparent surface should capture the ambient light information coming from the Zone Entity Key Light
SO as you use a SKymap or remove it you should see the ambient lighting of the surface change accordingly

**Comment by OmegaHeron** [comment]
Thanks - I'll give it a try in a bit.  Sounds like it will need some proper stage setup to really make impact so I'll wait a bit until I have ample tinker time.

**Comment by OmegaHeron** [comment]
Ok - it took a bit of tinkering with my zone settings, but, yes - I can see some coloration to transparency in test models with this PR while it looks uniformly same in color with production build.

File: libraries/render-utils/src/model_translucent.slf [hunk 1]
```diff
@@ -5,7 +5,7 @@
 //  model_translucent.frag
 //  fragment shader
 //
-//  Created by Andrzej Kapolka on 9/19/14.
+//  Created by Sam Gateau on 2/15/2016.
```
Comment (ZappoMan) [inline libraries/render-utils/src/model_translucent.slf]: :+1:

File: libraries/model/src/model/Light.slh [hunk 2]
```diff
@@ -11,6 +11,41 @@
...
+
+    vec4 value = C1 * sh.L22 * (dir.x * dir.x - dir.y * dir.y) + 
+                        C3 * sh.L20 * dir.z * dir.z + 
+                        C4 * sh.L00 - C5 * sh.L20 + 
+                        2.0 * C1 * (   sh.L2m2 * dir.x * dir.y + 
+                                    sh.L21  * dir.x * dir.z + 
+                                    sh.L2m1 * dir.y * dir.z ) + 
+                        2.0 * C2 * (   sh.L11  * dir.x + 
+                                    sh.L1m1 * dir.y + 
+                                    sh.L10  * dir.z ) ;
```
Comment (AndrewMeadows) [inline libraries/model/src/model/Light.slh]: I love this application of spherical harmonics.

Coding standard section 3.5.4 says that floating point literlals should have their trailing **f**.  Here in the value calculation and also in the coefficient declarations above.

https://docs.highfidelity.com/docs/coding-standard#section-3-5-4-when-using-a-single-precision-float-type-include-the-trailing-f-

**Pull request merged**